File: app/src/main/java/com/arcinteractive/spaces/data/model/SpaceInvite.kt
package com.arcinteractive.spaces.data.model

import java.util.Date

data class SpaceInvite(
    val id: String,
    val code: String,
    val spaceId: String,
    val spaceName: String,
    val spaceEmoji: String,
    val createdBy: String,
    val createdAt: Date,
    val expiresAt: Date,
    val maxUses: Int,
    val usedCount: Int,
    val active: Boolean
) {
    val remainingUses: Int
        get() = (maxUses - usedCount).coerceAtLeast(0)
}

File: iOS/Spaces V1.5/Models/SpaceInvite.swift
import Foundation

struct SpaceInvite: Identifiable, Hashable {
    let id: String
    let code: String
    let spaceId: String
    let spaceName: String
    let spaceEmoji: String
    let createdBy: String
    let createdAt: Date
    let expiresAt: Date
    let maxUses: Int
    let usedCount: Int
    let active: Bool

    var remainingUses: Int {
        max(0, maxUses - usedCount)
    }
}
